tests/framing-helper.c

```diff
@@ -51,9 +57,11 @@ static void remove_framing()
             break;
 
         // Check that the amount read matches the expected.
-        if (amount_read != amount_to_read)
-            errx(EXIT_FAILURE, "Expected to read %u bytes, but only got %u bytes.",
+        if (amount_read != amount_to_read) {
+            fprintf(stderr, "Expected to read %u bytes, but only got %u bytes.",
```

Comment:
This needs a `\n` at the end of the string. The other calls to `fprintf` need it as well. `err`, `errx`, etc. automatically add the `\n`.